cycle_error.go

```diff
@@ -46,6 +47,7 @@ func (e errCycleDetected) Error() string {
 	//
 	b := new(bytes.Buffer)
 
+	fmt.Fprintf(b, "In Scope %s: \n", e.scope.name)
```

Comment:
nit: I would do something like `[scope "foo"]` here so that the error is:

```
[scope "foo"]
foo provided by bar
  depends on baz provided by ...
```